packages/widgets/src/menu.ts

```diff
@@ -1371,6 +1378,41 @@ export namespace Menu {
       let kb = data.item.keyBinding;
       return kb ? CommandRegistry.formatKeystroke(kb.keys) : null;
     }
+
+    /**
+     * @returns The aria label content to add to the shortcut node.
+     */
+    formatShortcutText(data: IRenderData): h.Child {
+      const keyToText: { [key: string]: string } = {
+        ']': 'Closing bracket',
+        '[': 'Opening bracket',
+        ',': 'Comma',
+        '.': 'Full stop',
+        "'": 'Single quote',
+        '-': 'Hyphen-minus'
+      };
+
+      let kbText = data.item.keyBinding;
+      let result = kbText ? CommandRegistry.formatKeystroke(kbText.keys) : null;
+
+      let punctuationRegex = /\p{P}/u;
+      let punctuations = result?.match(punctuationRegex);
+      if (!punctuations) {
+        return [];
+      }
+      for (const punctuation of punctuations) {
+        if (result != null && Object.keys(keyToText).includes(punctuation)) {
+          const individualKeys = result.split('+');
+          let index = individualKeys.indexOf(punctuation);
+          if (index != -1) {
+            individualKeys[index] = keyToText[punctuation];
+          }
+          const textShortcut = individualKeys.join('+');
+          return textShortcut;
+        }
+      }
+      return kbText ? CommandRegistry.formatKeystroke(kbText.keys) : null;
```

Comment:
All right, this makes it pretty clear that we need to have this implemented at `CommandRegistry` level (see earlier suggestion on creating `CommandRegistry.formatKeystrokeAriaLabel`)

Reply:
Hi @krassowski, 

All makes sense, thank you for your comments. I will revert back to draft for now whist I work on these and let you know once ready for review again.